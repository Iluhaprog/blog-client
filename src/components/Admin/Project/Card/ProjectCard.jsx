import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import {Button, Container, Row} from 'react-bootstrap';
import {Github, ShareFill} from 'react-bootstrap-icons';
import broken from '../../../../assets/brkImg.png';
import {themes} from '../../../../store/settings/settingsReducer';

const ProjectCardBox = styled.div`
  position: relative;
  max-width: 350px;
  width: 100%;
  height: 300px;
  border-radius: 10px;
  margin: 15px 0;
  box-shadow: 0 0 15px rgba(0, 0, 0, .3);
  overflow: hidden;
  
  &:hover {
    div:last-child {
      top: 0;
    }
  }
`;
const ProjectPreview = styled.div`
  position: absolute;
  width: 100%;
  height: 300px;
  top: 0;
  left: 0;
  
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    object-position: center;
  }
`;
const ProjectDescription = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  border-radius: 10px;
  padding: 10px 30px;
  top: calc(100% - 50px);
  transition: .2s ease-in-out;
  box-shadow: 0 -10px 10px rgba(0, 0, 0, .2);
  background-color: ${({theme}) => {
    return theme === themes.DARK ? '#2a2c32' : '#fff';
  }};
  
  p {
    max-height: 110px;
    overflow-y: scroll;
  }
`;

const ProjectCard = (props) => {
  const {preview, title, description, projectLink, githubLink} = props;
  const {theme, isAdmin, lang} = props;
  const {onClick, onDelete} = props;
  const API_URL = process.env.REACT_APP_API_URL;

  return (
    <ProjectCardBox>
      <ProjectPreview>
        <img
          src={preview ? `${API_URL}/${preview}` : broken}
          alt={preview}
        />
      </ProjectPreview>
      <ProjectDescription theme={theme}>
        <h3>{title}</h3>
        <p>{description}</p>
        <Container>
          <Row>
            <Button
              variant={theme}
              style={{marginRight: '10px'}}
              onClick={() => window.open(projectLink, '_blank')}
            >
              <ShareFill />
            </Button>
            <Button
              variant={theme}
              onClick={() => window.open(githubLink, '_blank')}
            >
              <Github />
            </Button>
          </Row>
          {
            isAdmin && (
              <>
                <hr />
                <Row className={'justify-content-between'}>
                  <Button
                    variant={'primary'}
                    onClick={() => onClick()}
                  >
                    {lang.button.EDIT}
                  </Button>
                  <Button
                    variant={'danger'}
                    onClick={() => onDelete()}
                  >
                    {lang.button.DELETE}
                  </Button>
                </Row>
              </>
            )
          }
        </Container>
      </ProjectDescription>
    </ProjectCardBox>
  );
};

ProjectCard.propTypes = {
  theme: PropTypes.string,
  isAdmin: PropTypes.bool,
  lang: PropTypes.object,
  preview: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  projectLink: PropTypes.string,
  githubLink: PropTypes.string,
  onClick: PropTypes.func,
  onDelete: PropTypes.func,
};

export {ProjectCard};
