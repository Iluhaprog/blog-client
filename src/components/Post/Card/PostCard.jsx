import React from 'react';
import PropTypes from 'prop-types';
import {Badge, Button, Card, Col, Container, Row} from 'react-bootstrap';
import broken from '../../../assets/brkImg.png';
import {Eye, EyeSlash} from 'react-bootstrap-icons';
import {useHistory} from 'react-router';

const PostCard = ({post, lang, theme, onDelete}) => {
  const API_URL = process.env.REACT_APP_API_URL;
  const history = useHistory();
  return (
    <Card
      bg={theme}
      style={{width: '300px'}}
    >
      <Card.Img
        variant={'top'}
        style={{
          width: '100%',
          height: '200px',
          objectFit: 'cover',
          objectPosition: 'center',
        }}
        src={post.preview ? `${API_URL}/${post.preview}` : broken}
      />
      <Card.Body>
        <Card.Title>
          <Container>
            <Row className='justify-content-between'>
              {post.title.length > 20 ?
                  post.title?.slice(0, 20) + '...' :
                  post.title
              }
              <Badge
                variant='secondary'
              >
                {
                  post.isVisible ? <Eye /> : <EyeSlash />
                }
              </Badge>
            </Row>
          </Container>
        </Card.Title>
        <Card.Text>
          {post.description.length > 100 ?
              post.description.slice(0, 100) + '...' :
              post.description
          }
        </Card.Text>
        <Row>
          <Col md={12}>
            {
              post.tags?.map((tag) => (
                <Badge
                  key={tag.id}
                  variant={'secondary'}
                  style={{marginRight: '5px'}}
                >
                  {tag.title}
                </Badge>
              ))
            }
          </Col>
        </Row>
      </Card.Body>
      <Card.Footer>
        <Row className='justify-content-between align-items-center'>
          <Col md={3}>
            <Button
              variant={'primary'}
              onClick={() => history.push(`/admin/post/${post.id}`)}
            >
              {lang.button.READ}
            </Button>
          </Col>
          <Col md={3}>
            <Button
              onClick={() => onDelete()}
              variant={'danger'}
            >
              {lang.button.DELETE}
            </Button>
          </Col>
          <Col md={6}>
            <time
              className='text-muted'
              style={{
                paddingLeft: '30px',
              }}
            >
              {post.creationDate?.split('T')[0]}
            </time>
          </Col>
        </Row>
      </Card.Footer>
    </Card>
  );
};

PostCard.propTypes = {
  post: PropTypes.object,
  theme: PropTypes.string,
  lang: PropTypes.object,
  onDelete: PropTypes.func,
};

export {PostCard};
