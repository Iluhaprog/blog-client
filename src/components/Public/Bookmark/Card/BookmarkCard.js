import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {Row} from '../../Row';
// import {Image} from '../../UI/Image';
import {getEntityDataByLang} from '../../../../utils/data/data';
import {Title} from '../../UI/Title';
import {Separator} from '../../Separator';
import {Button} from '../../UI/Button';
import {Trash} from 'react-bootstrap-icons';
import {NavLink} from 'react-router-dom';

const Card = styled.div`
  max-width: 300px;
  min-width: 150px;
  width: 100%;
  margin: 10px;
  border: 1px solid #000;
  border-radius: 5px;
  padding: 10px;
  
  a {
    text-decoration: none;
    color: #000;
    
    h2 {
      margin: 0;
      font-size: 20px;
    }
  }
`;

const BookmarkCard = ({data, lang, onDelete}) => {
  data = getEntityDataByLang(data, lang.title, 'postData');
  const API_URL = process.env.REACT_APP_API_URL;

  return (
    <Card>
      <Row alignItems={'center'} justifyContent={'space-between'}>
        <Row alignItems={'center'}>
          <img
            width={50}
            height={50}
            src={`${API_URL}/${data.preview}`}
            alt={data.title}
          />
          <Separator indentLeft={20} />
          <NavLink to={`/post/${data.id}`}>
            <Title>
              {data.title}
            </Title>
          </NavLink>
        </Row>
        <Button
          onClick={onDelete}
          minWidth={20}
          mod={'small'}
        >
          <Trash />
        </Button>
      </Row>
    </Card>
  );
};

BookmarkCard.propTypes = {
  data: PropTypes.object,
  lang: PropTypes.object,
  onDelete: PropTypes.func,
};

export {BookmarkCard};
