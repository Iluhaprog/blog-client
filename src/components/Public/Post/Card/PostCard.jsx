import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {Image} from '../../UI/Image';
import empty from '../../../../assets/brkImg.png';
import {Title} from '../../UI/Title';
import {Description} from '../../UI/Description';
import {Time} from '../../UI/Time';
import {getDateAndTime} from '../../../../utils/string/string';
import {Row} from '../../Row';
import {Tag} from '../../UI/Tag';
import {Button} from '../../UI/Button';
import {Separator} from '../../Separator';
import {Bookmark} from 'react-bootstrap-icons';

const Card = styled.article`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 50px;
  
  &:last-child {
    margin-bottom: 0;
  }
  
  figure {
    margin: 0;
  }
  
  footer {
    padding-left: 5px;
  }
  
  @media screen and (max-width: 420px) {
    figure {
      margin: 0 auto;
      max-width: 100%;
      img {
        width: 100%;
      }
    }
    
    footer {
      padding: 0 20px 0 0;
    }
  }
`;

const CardContent = styled.section`
  max-width: 500px;
  width: 100%;
  padding: 10px 0 10px 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

const PostCard = (props) => {
  const {title, preview, description, creationDate, tags, lang} = props;
  const {onClick, addToBookmarks} = props;
  const API_URL = process.env.REACT_APP_API_URL;
  const [date, time] = getDateAndTime(creationDate || '');

  return (
    <Card>
      <Image
        width={350}
        height={350}
      >
        <img src={preview ? API_URL + '/' + preview : empty} alt={title}/>
      </Image>
      <CardContent>
        <div>
          <header>
            <Title>
              {title}
            </Title>
          </header>
          <Description>
            {description}
          </Description>
          <Time>
            {`${date} - ${time}`}
          </Time>
        </div>
        <div>
          <Row justifyContent={'flex-start'} wrap={'wrap'}>
            {
              tags?.map((tag) => (
                <Tag key={tag.id} title={tag.title} />
              ))
            }
          </Row>
          <Separator indentTop={10} />
          <footer>
            <Row justifyContent={'space-between'}>
              <Button
                onClick={onClick}
                mod={'small'}
              >
                {lang.button.READ}
              </Button>
              {addToBookmarks && (
                <Button
                  onClick={addToBookmarks}
                  mod={'small'}
                  minWidth={20}
                >
                  <Bookmark/>
                </Button>
              )
              }
            </Row>
          </footer>
        </div>
      </CardContent>
    </Card>
  );
};

PostCard.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  preview: PropTypes.string,
  tags: PropTypes.array,
  creationDate: PropTypes.string,
  lang: PropTypes.object,
  onClick: PropTypes.func,
  addToBookmarks: PropTypes.func,
};

export {PostCard};
