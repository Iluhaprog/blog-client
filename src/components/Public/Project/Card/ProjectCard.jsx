import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {Title} from '../../UI/Title';
import {Description} from '../../UI/Description';
import empty from '../../../../assets/brkImg.png';
import {Row} from '../../Row';
import {ReactComponent as Link} from '../../../../assets/public/icons/link.svg';
import {
  ReactComponent as Github,
} from '../../../../assets/public/icons/github.svg';

const ProjectCardBox = styled.article`
  position: relative;
  max-width: 350px;
  width: 100%;
  height: 350px;
  overflow: hidden;
  border: 1px solid var(--color-1);
  margin: 10px
`;

const ProjectCardPreview = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  
  img {
    width: 100%;
    height: 100%;
    object-position: center;
    object-fit: cover;
  }
`;

const ProjectCardItem = styled.div`
  position: absolute;
  top: 75%;
  width: 100%;
  height: 100%;
  padding: 20px 20px;
  background-color: #fff;
  transition: .2s ease-in-out;
  
  &:after, &:before {
    content: '';
    display: block;
    width: 30px;
    height: 1px;
    background-color: var(--color-1);
    margin: 0 0 15px 0;
  }
  
  &:after {
    margin-top: 15px;  
  }
  
  &:hover {
    top: 0;
  }
  
  a {
    margin-right: 10px;
    
    svg  {
      width: 30px;
      height: 30px;
    }
  }
`;

const ProjectCard = (props) => {
  const {title, description, preview, githubLink, projectLink} = props;
  const API_URL = process.env.REACT_APP_API_URL;

  return (
    <ProjectCardBox>
      <ProjectCardPreview>
        <img src={preview ? API_URL + '/' + preview : empty} alt={title} />
      </ProjectCardPreview>
      <ProjectCardItem>
        <Title>{title}</Title>
        <Description>
          {description}
        </Description>
        <Row>
          <a href={projectLink}><Link /></a>
          <a href={githubLink}><Github /></a>
        </Row>
      </ProjectCardItem>
    </ProjectCardBox>
  );
};

ProjectCard.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  projectLink: PropTypes.string,
  githubLink: PropTypes.string,
  preview: PropTypes.string,
};

export {ProjectCard};
