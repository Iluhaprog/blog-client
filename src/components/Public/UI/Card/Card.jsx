import styled from 'styled-components';
import PropTypes from 'prop-types';

const Card = styled.div`
  max-width: ${({maxWidth}) => `${maxWidth}px` || '500px'};
  max-height: ${({maxHeight}) => `${maxHeight}px` || '500px'};
  width: 100%;
  height: 100vh;
  background-color: #fff;
  padding: 15px;
`;

Card.propTypes = {
  maxWidth: PropTypes.number,
  maxHeight: PropTypes.number,
};

export {Card};
