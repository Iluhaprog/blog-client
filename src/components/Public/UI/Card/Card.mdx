---
name: Card
route: /card
---

import {Playground, Props} from 'docz';
import {Card} from './Card';

<Props of={Card} />

# Basic usage
<Playground>
  <Card maxWidth={200} maxHeight={200}>
    Card content
  </Card>
</Playground>
