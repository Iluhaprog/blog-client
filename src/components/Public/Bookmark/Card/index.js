import {BookmarkCard} from './BookmarkCard';
export {BookmarkCard};
