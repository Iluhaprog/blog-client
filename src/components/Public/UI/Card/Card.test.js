import {render} from '@testing-library/react';
import React from 'react';
import {Card} from './Card';

test('Should render Background component', () => {
  const {asFragment} = render(
    <Card maxWidth={100} maxHeight={100}>
      Hello
    </Card>
  );
  expect(asFragment()).toMatchSnapshot();
});
