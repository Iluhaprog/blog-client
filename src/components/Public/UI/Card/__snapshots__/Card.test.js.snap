// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Background component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ eppQqW"
  >
    Hello
  </div>
</DocumentFragment>
`;
